app/controllers/api/connect/v3/systems/products_controller.rb

```diff
@@ -10,16 +10,17 @@ def activate
   end
 
   def show
-    unless @system.products.include? @product
+    if @system.products.include? @product
+      respond_with(
+        @product,
+        serializer: ::V3::ProductSerializer,
+        uri_options: { scheme: request.scheme, host: request.host, port: request.port },
+        service_url: url_for(controller: '/services', action: :show, id: @product.service.id)
+      )
+    else
       untranslated = N_("The requested product '%s' is not activated on this system." % @product.friendly_name)
-      respond_with_error({ message: untranslated, localized_message: _(untranslated) }) and return
+      raise ActionController::TranslatedError.new(error: untranslated, localized_error: _(untranslated))
```

Comment:
That actually seems to be incorrect, because string interpolation happens inside of N_().